source/extensions/filters/http/oauth2/filter.cc

```diff
@@ -289,10 +290,14 @@ Http::FilterHeadersStatus OAuth2Filter::decodeHeaders(Http::RequestHeaderMap& he
     }
   }
 
-  // Sanitize the Authorization header, since we have no way to validate its content. Also,
-  // if token forwarding is enabled, this header will be set based on what is on the HMAC cookie
-  // before forwarding the request upstream.
-  headers.removeInline(authorization_handle.handle());
+  // preserve the existing authorization header when forward_bearer_token is false and
+  // preserve_authorization_header is true
+  if (config_->forwardBearerToken() == true || config_->preserveAuthorizationHeader() == false) {
```

Comment:
If we do the config validation when we load the configuration. Then you only need check the `preserveAuthorizationHeader()` here. And we needn't to check bool by `== true` or `== false`.